language/tools/move-cli/src/sandbox/utils/package.rs

```diff
@@ -167,18 +168,21 @@ impl MovePackage {
         Ok(src_dirs)
     }
 
-    pub(crate) fn compiled_modules(&self, out_path: &Path) -> Result<Vec<CompiledModule>> {
+    pub(crate) fn compiled_modules(
+        &self,
+        out_path: &Path,
+    ) -> Result<Vec<(ModuleIdWithNamedAddress, CompiledModule)>> {
         let mut modules = vec![];
         for dep in self.deps.iter() {
             modules.extend(dep.compiled_modules(out_path)?);
         }
         for entry in find_filenames(&[path_to_string(&self.get_binary_dir(out_path))?], |path| {
             extension_equals(path, MOVE_COMPILED_EXTENSION)
         })? {
-            modules.push(
-                CompiledModule::deserialize(&fs::read(Path::new(&entry)).unwrap())
-                    .map_err(|e| anyhow!("Failure deserializing module {}: {:?}", entry, e))?,
-            );
+            let module = CompiledModule::deserialize(&fs::read(Path::new(&entry)).unwrap())
+                .map_err(|e| anyhow!("Failure deserializing module {}: {:?}", entry, e))?;
+            // TODO support named addresses
```

Comment:
Small rant: Personally I like `TODO` with more information about how to approach the problem, but I guess anyone who comes across this can just `git blame` and find out what "named addresses" means.

Although, now that you mention it, what's preventing you from adding support as part of this pull request? Is it because you'd need to thread this new interface metadata through the `CompiledModule::deserialize` method, and that would be a big diff?

Reply:
The problem with adding such support is basically replicating and building an entire package system. Closer to what @tzakian is doing. My thought is support that here with Tim's build system. So maybe I should comment that

Reply:
Agree with Todd here -- I think this is something that we'll want the build system to support, since it will need to support this type of dealing with named addresses and deserializing modules more generally. Eventually once we have proper packages the Move CLI should be dealing with modules through that then.